File: 315 hw1/Codes/duru_ant.rb

# my method to print
def foo(k, v)
    puts "key: #{k} , value: #{v}"
end


# Initializing the array
myHash = {"first" => 7, "second" => 19, "third" => 63, "fourth" => 81, "fifth" => 109}
puts "------1-Hash has been initialized.-------"


# Getting Value for a given key
puts "------2-Getting the value of key \"first\".-------" 
myValue = myHash["first"]
puts "the value with key \"first\" is: #{myHash["first"]}"

# Adding a new element
myHash["sixth"] = 6
puts "------3-Element with key \"sixth\" and value 6 is added.-------"


# Removing an element
myHash.delete("third")
puts "------4-Element with key \"third\" is deleted.-------"

# Modify the value of an element
puts "------5-Modifying value of an element.-------"
puts "The value of the element with key \"fifth\" is: #{myHash["fifth"]}"
myHash["fifth"] = 5
puts "After modification the element with key \"fifth\" is: #{myHash["fifth"]}"

# Search for existence of a key
puts "------6-Searching whether key \"second\" is in the hash.-------"
if myHash["third"] != nil
    puts "the key \"third\" is in the hash with value: #{myHash["third"]}"
else
    puts "the key \"third\" is not in the hash"

end


# Search for existence of a value
puts "------7-Searching whether value 81 is in the hash.-------"
if myHash.key(81) != nil
    puts "81 is in the hash with key: #{myHash.key(81)}"
else
    puts "81 is not in the hash"

end

# Loop
puts "------8-Iterating through hash and printing pairs.-------"
myHash.each do |key, value|
    foo(key, value)
end

File: 315 hw1/Codes/duru_ant.dart
void main(){
   // Initializing the hash
   Map myMap = {
                 'first': 1,
                 'second': 2,
                 'third': 3,
                 'fourth': 4,
                 'fifth': 5};
   print("-----------1-Associative Array Has Been Initialized.-----------");
   
   // Getting value by a given key
   print("-----------2-Getting the value of an element.-----------");
   var myValue = myMap["first"];
   print("value with the key \"first\" is:  $myValue ");
   
   // Adding an element
   print("-----------3-Adding an element.-----------");
   myMap["sixth"] = 6;
   print("An element with key \"sixth\" and value 6 has been added");
   
   // Removing an element
   print("-----------4-Removing an element.-----------");
   myMap.remove('third');
   print("The element with key \"third\" has been removed");
   
   // Modifying an element
   print("-----------5-Modifying an element.-----------");
   var before = myMap['fourth'];
   print("Before modify, the value with key \"fourth\" was: $before");
   myMap["fourth"] = 7;
   var after = myMap['fourth'];
   print("After modify, the value with key \"fourth\" is: $after");
   
   // Search for existence of a key
   print("-----------6-Searching for existence of a key.-----------");
   if(myMap["first"] != null){
       var fValue = myMap['first'];
       print("the key \"first\" is in the hash with value: $fValue ");
   }
   else{
       print("the key \"first\" is not in the hash");
   }
   
   // Search for existence of a value
   print("-----------7-Searching for existence of a value.-----------");
   var e = myMap.keys.firstWhere((k) => myMap[k] == 2, orElse: () => null);
   
   if(e == null)
   {
       print("the value 2 is not in the hash");
   }
   else{
       print("the value 2 is in the hash");
   }
   
   // Loop through the hash and print pairs
   print("-----------8-Iterating through the hash and printing pairs.-----------");
   void foo(k, v){
       print("key: $k , value: $v");
   }
   
   myMap.forEach((k, v) {
    foo(k,v);
    });
}